File: 2020/day8/part1/main_test.py
from typing import Tuple
from unittest import TestCase

from advent_of_code.day8.part1.main import action, main

class TestDay8Part1(TestCase):

    def test_action(self) -> None:
        expected_result1: Tuple[int] = 1, 0
        expected_result2: Tuple[int] = -3, 0
        expected_result3: Tuple[int] = 1, 6
        self.assertEqual(expected_result1, action(row="nop +0"))
        self.assertEqual(expected_result2, action(row="jmp -3"))
        self.assertEqual(expected_result3, action(row="acc +6"))

    def test_main(self) -> None:
        self.assertEqual(main(filepath="tests/day8/input_test.txt"), 5)

File: 2020/day8/part1/main_test.java
package tests.day8.part1;

import advent_of_code.day8.part1.*;

import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;

import static org.junit.jupiter.api.Assertions.assertEquals;
import static org.junit.jupiter.api.Assertions.assertTrue;
import static org.junit.jupiter.api.Assertions.assertFalse;

import org.junit.jupiter.api.DisplayName;
import org.junit.jupiter.api.Test;

public class main_test {
    
    @Test                                               
    @DisplayName("change")   
    public void testAction() {
        assertEquals(part1.Action("nop +0"), Arrays.asList(1, 0));
        assertEquals(part1.Action("jmp -3"), Arrays.asList(-3, 0));
        assertEquals(part1.Action("acc +6"), Arrays.asList(1, 6));
    }

    @Test
    @DisplayName("main")
    public void testMain() throws Exception {
        assertEquals(5, part1.main("tests/day8/input_test.txt"));
    }

}